core/directory.go

```diff
@@ -1136,6 +1136,52 @@ func (dir *Directory) Mount(ctx context.Context, f func(string) error) error {
 	})
 }
 
+func (dir *Directory) GitIgnoreFor(ctx context.Context) ([]string, error) {
+	result := []string{}
+
+	_, err := execInMount(ctx, dir, func(root string) error {
+		resolvedDir, err := containerdfs.RootPath(root, dir.Dir)
+		if err != nil {
+			return err
+		}
+
+		return filepath.WalkDir(resolvedDir, func(path string, d fs.DirEntry, prevErr error) error {
+			if prevErr != nil {
+				return prevErr
+			}
+
+			if d.Name() != ".gitignore" {
```

Comment:
Make sure `d` is a regular file. If a directory is called `.gitignore` (for some reason), we shouldn't error out.

Reply:
Applied in [7ff0141](https://github.com/dagger/dagger/pull/10853/commits/7ff0141df57caa39e12fadd9c25b19c5062431bb)